src/mac.c

```diff
@@ -185,3 +185,181 @@ static int handle_lbt_mode(struct nl802154_state *state,
 }
 COMMAND(set, lbt, "<1|0>",
 	NL802154_CMD_SET_LBT_MODE, 0, CIB_NETDEV, handle_lbt_mode, NULL);
+
+static int print_association_confirm_handler(struct nl_msg *msg, void *arg)
+{
+	int r;
+
+	uint16_t assoc_short_addr;
+	uint8_t status;
+
+	struct genlmsghdr *gnlh;
+	struct nlattr *tb[ NL802154_ATTR_MAX + 1 ];
+	int i;
+
+	printf( "in %s\n", __FUNCTION__ );
+	int i,j;
+
+	gnlh = nlmsg_data( nlmsg_hdr( msg ) );
+	if ( NULL ==  gnlh ) {
+		fprintf( stderr, "gnlh was null\n" );
+		goto protocol_error;
+	}
+
+	r = nla_parse( tb, NL802154_ATTR_MAX, genlmsg_attrdata( gnlh, 0 ),
+	genlmsg_attrlen( gnlh, 0 ), NULL );
+	if ( 0 != r ) {
+	fprintf( stderr, "nla_parse\n" );
+	goto protocol_error;
+	}
+
+	if ( ! (
+		tb[ NL802154_ATTR_SHORT_ADDR ] &&
+		tb[ NL802154_ATTR_CONFIRM_STATUS ]
+	) ){
+		r = -EINVAL;
+		goto out;
+	}
+
+	assoc_short_addr = nla_get_u16( tb[ NL802154_ATTR_SHORT_ADDR ] );
+	status = nla_get_u8( tb[ NL802154_ATTR_CONFIRM_STATUS ] );
+
+	printf(
+		"associated short address: %u, "
+		"confirm status: %u, ",
+		assoc_short_addr,
+		status
+	);
+
+protocol_error:
+	fprintf( stderr, "protocol error\n" );
+	r = -EINVAL;
+
+out:
+	printf( "returning %d\n", r );
+	return r;
+}
+
+static int handle_set_association_request(struct nl802154_state *state,
+	       struct nl_cb *cb,
+	       struct nl_msg *msg,
+	       int argc, char **argv,
+	       enum id_input id)
+{
+	int r;
+
+	uint8_t coord_channel;
+	uint8_t coord_page;
+	enum nl802154_address_modes addr_mode;
+	uint16_t coord_pan_id;
+	uint64_t coord_addr;
+	uint8_t capability_info;
+
+	/* CHANNEL */
+	if ( argc >= 1 ){
+		if ( 1 != sscanf( argv[ 0 ], "%u", &coord_channel ) ) {
+			goto invalid_arg;
+		}
+	}
+	/* PAGE */
+	if ( argc >= 2 ){
+		if ( 1 != sscanf( argv[ 1 ], "%u", &coord_page ) ) {
+			goto invalid_arg;
+		}
+	}
+	/* ADDR MODE */
+	if ( argc >= 3 ){
+		if ( 1 != sscanf( argv[ 2 ], "%u", &addr_mode ) ) {
+			goto invalid_arg;
+		}
+	}
+	/* PAN ID */
+	if ( argc >= 4 ){
+		if ( 1 != sscanf( argv[ 3 ], "%u", &coord_pan_id ) ) {
+			goto invalid_arg;
+		}
+	}
+	/* ADDRESS */
+	if ( argc >= 5 ){
+		if ( 1 != sscanf( argv[ 4 ], "%u", &coord_addr ) ) {
+			goto invalid_arg;
+		}
+	}
+
+	/* CAPABILITY INFO */
+	if ( argc == 6 ){
+		if ( 1 != sscanf( argv[ 5 ], "%u", &capability_info ) ) {
+			goto invalid_arg;
+		}
+	}
+	if ( 0 != r ){
+		goto out;
+	}
+
+	NLA_PUT_U8(msg, NL802154_ATTR_CHANNEL, coord_channel);
+	NLA_PUT_U8(msg, NL802154_ATTR_PAGE, coord_page);
+	NLA_PUT_U8(msg, NL802154_ATTR_ADDRESS_MODE, addr_mode);
+	NLA_PUT_U16(msg, NL802154_ATTR_PAN_ID, htole16(coord_pan_id));
+	if ( NL802154_ADDR_SHORT == addr_mode ){
```

Comment:
I have a question for this section, should I use a UNION type for coord_addr, as it can be either short or the extended address, or is it fine as-is?